src/plugins/discover/public/application/apps/main/components/sidebar/discover_field_details.test.tsx

```diff
@@ -25,18 +25,19 @@ const indexPattern = getStubIndexPattern(
 );
 
 describe('discover sidebar field details', function () {
+  const onAddFilter = jest.fn();
   const defaultProps = {
     indexPattern,
     details: { buckets: [], error: '', exists: 1, total: 2, columns: [] },
-    onAddFilter: jest.fn(),
+    onAddFilter,
   };
 
   function mountComponent(field: IndexPatternField) {
     const compProps = { ...defaultProps, field };
     return mountWithIntl(<DiscoverFieldDetails {...compProps} />);
   }
 
-  it('should enable the visualize link for a number field', function () {
```

Comment:
:information_source: I've removed all those tests, since they were anyway not working. The only test that checked if the button was visible, did not do that correctly (since the result of `findTestSubject` will always be truthy and a `.exists()` check is required), and also did not work, due to the async nature of the component. I instead decided to replace this by unit tests for the utility class to load that visualization information.